File: modules/apps/object/object-admin-rest-client-js/src/models/ObjectFolderItem.ts
/**
 * SPDX-FileCopyrightText: (c) 2025 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

			import {ObjectDefinition} from './ObjectDefinition';

/**
 * @author Javier Gamarra
 * @generated
 */

	export class ObjectFolderItem {
			"linkedObjectDefinition"?: boolean;
			"objectDefinition"?: ObjectDefinition;
			"objectDefinitionExternalReferenceCode"?: string;
			"positionX"?: number;
			"positionY"?: number;

		static "discriminator": string | undefined = undefined;

	static "attributeTypeMap": Array<{
		baseName: string;
		name: string;
		type: string;
	}> = [
		{
			baseName: "linkedObjectDefinition",
			name: "linkedObjectDefinition",
			type: "boolean",
		},
		{
			baseName: "objectDefinition",
			name: "objectDefinition",
			type: "ObjectDefinition",
		},
		{
			baseName: "objectDefinitionExternalReferenceCode",
			name: "objectDefinitionExternalReferenceCode",
			type: "string",
		},
		{
			baseName: "positionX",
			name: "positionX",
			type: "number",
		},
		{
			baseName: "positionY",
			name: "positionY",
			type: "number",
		},
		];

		static getAttributeTypeMap() {
				return ObjectFolderItem.attributeTypeMap;
		}
	}

File: modules/apps/object/object-api/src/main/java/com/liferay/object/model/ObjectFolderItem.java
/**
 * SPDX-FileCopyrightText: (c) 2023 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

package com.liferay.object.model;

import com.liferay.portal.kernel.annotation.ImplementationClassName;
import com.liferay.portal.kernel.model.PersistedModel;
import com.liferay.portal.kernel.util.Accessor;

import org.osgi.annotation.versioning.ProviderType;

/**
 * The extended model interface for the ObjectFolderItem service. Represents a row in the &quot;ObjectFolderItem&quot; database table, with each column mapped to a property of this class.
 *
 * @author Marco Leo
 * @see ObjectFolderItemModel
 * @generated
 */
@ImplementationClassName("com.liferay.object.model.impl.ObjectFolderItemImpl")
@ProviderType
public interface ObjectFolderItem
	extends ObjectFolderItemModel, PersistedModel {

	/*
	 * NOTE FOR DEVELOPERS:
	 *
	 * Never modify this interface directly. Add methods to <code>com.liferay.object.model.impl.ObjectFolderItemImpl</code> and rerun ServiceBuilder to automatically copy the method declarations to this interface.
	 */
	public static final Accessor<ObjectFolderItem, Long>
		OBJECT_FOLDER_ITEM_ID_ACCESSOR =
			new Accessor<ObjectFolderItem, Long>() {

				@Override
				public Long get(ObjectFolderItem objectFolderItem) {
					return objectFolderItem.getObjectFolderItemId();
				}

				@Override
				public Class<Long> getAttributeClass() {
					return Long.class;
				}

				@Override
				public Class<ObjectFolderItem> getTypeClass() {
					return ObjectFolderItem.class;
				}

			};

}
// SB-Hash:-1587391558